autonomy/chain/tx.py

```diff
@@ -100,288 +106,136 @@ def __init__(
         self.chain_type = chain_type
         self.ledger_api = ledger_api
         self.crypto = crypto
-        self.tx = None
+        self.tx_builder = tx_builder
         self.timeout = timeout or DEFAULT_ON_CHAIN_INTERACT_TIMEOUT
         self.retries = retries or DEFAULT_ON_CHAIN_INTERACT_RETRIES
         self.sleep = sleep or DEFAULT_ON_CHAIN_INTERACT_SLEEP
 
-    def build(
-        self,
-        method: Callable[[], Dict],
-        contract: str,
-        kwargs: Dict,
-    ) -> Dict:
-        """Build transaction."""
-        return method(  # type: ignore
-            ledger_api=self.ledger_api,
-            contract_address=ContractConfigs.get(name=contract).contracts[
-                self.chain_type
-            ],
-            raise_on_try=True,
-            **kwargs,
-        )
-
     def _reprice(self, tx_dict: Dict) -> Optional[Dict]:
         """Reprice transaction."""
-        if "maxFeePerGas" not in tx_dict or "maxPriorityFeePerGas" not in tx_dict:
+        if "maxFeePerGas" in tx_dict and "maxPriorityFeePerGas" in tx_dict:
+            gas_price_strategy = "eip1559"
```

Comment:
This could be an Enum, but it's fine for now. Consider if it's worthwile adding it.